sale_order_general_discount/models/sale_order_line.py

```diff
@@ -17,7 +18,11 @@ def create(self, vals):
         """Apply general discount for sale order lines which are not created
         from sale order form view.
         """
-        if "discount" not in vals and "order_id" in vals:
+        if (
+            not ("flag" in self.env.context and self.env.context["flag"])
```

Comment:
Thanks @aleuffre, I added changes that you suggested me. It doesn't depends anymore from https://github.com/OCA/sale-workflow/pull/2668 because all changes are here. I think I need to close that Pr